src/osd/OSD.cc

```diff
@@ -8185,6 +8224,8 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb )
     }
   }
   pair<PGRef, PGQueueable> item = sdata->pqueue.dequeue();
+  sdata->ql_accum += sdata->pqueue.empty() ? 1 : 2; 
```

Comment:
ql_accum is supposed to be the sum of all queue lengths at dispatch time for this shard.  The measurement on a per-shard basis allows you to calculate the distribution of work across shards.  Sometimes we've suspected that one shard gets more backed up than another.  So that is why it's calculated here by each shard's queue length.      To calculate average queue length this number is divided by the number of dispatches that have happened.   Let me know if this doesn't answer the question.